vertica/datadog_checks/vertica/vertica.py

```diff
@@ -0,0 +1,485 @@
+
+from . import views
+from .utils import node_state_to_service_check
+
+# Python 3 only
+PROTOCOL_TLS_CLIENT = getattr(ssl, 'PROTOCOL_TLS_CLIENT', ssl.PROTOCOL_TLS)
+
+
+class VerticaCheck(AgentCheck):
+    __NAMESPACE__ = 'vertica'
+    SERVICE_CHECK_CONNECT = 'can_connect'
+    SERVICE_CHECK_NODE_STATE = 'node_state'
+
+    def __init__(self, name, init_config, instances):
+        super(VerticaCheck, self).__init__(name, init_config, instances)
+
+        self._db = self.instance.get('db', '')
+        self._server = self.instance.get('server', '')
+        self._port = self.instance.get('port', 5433)
+        self._username = self.instance.get('username', '')
+        self._password = self.instance.get('password', '')
+        self._backup_servers = [
+            '{}:{}'.format(bs.get('server', self._server), bs.get('port', self._port))
+            for bs in self.instance.get('backup_servers', [])
+        ]
+        self._connection_load_balance = is_affirmative(self.instance.get('connection_load_balance', False))
+        self._timeout = float(self.instance.get('timeout', 10))
+        self._tags = self.instance.get('tags', [])
+
+        self._tls_verify = is_affirmative(self.instance.get('tls_verify', False))
+        self._validate_hostname = is_affirmative(self.instance.get('validate_hostname', True))
+
+        self._cert = self.instance.get('cert', '')
+        if self._cert:  # no cov
+            self._cert = expanduser(self._cert)
+            self._tls_verify = True
+
+        self._private_key = self.instance.get('private_key', '')
+        if self._private_key:  # no cov
+            self._private_key = expanduser(self._private_key)
+
+        self._cafile = None
+        self._capath = None
+        ca_cert = self.instance.get('ca_cert', '')
+        if ca_cert:  # no cov
+            ca_cert = expanduser(ca_cert)
+            if isdir(ca_cert):
+                self._capath = ca_cert
+            else:
+                self._cafile = ca_cert
+
+            self._tls_verify = True
+
+        custom_queries = self.instance.get('custom_queries', [])
+        use_global_custom_queries = self.instance.get('use_global_custom_queries', True)
+
+        # Handle overrides
+        if use_global_custom_queries == 'extend':
+            custom_queries.extend(self.init_config.get('global_custom_queries', []))
+        elif 'global_custom_queries' in self.init_config and is_affirmative(use_global_custom_queries):
+            custom_queries = self.init_config.get('global_custom_queries', [])
+
+        # Deduplicate
+        self._custom_queries = list(iter_unique(custom_queries))
+
+        # Add global database tag
+        self._tags.append('db:{}'.format(self._db))
+
+        # We'll connect on the first check run
+        self._conn = None
+
+        # Cache database results for re-use among disparate functions
+        self._view = defaultdict(list)
+
+    def check(self, instance):
```

Comment:
Can we completely ignore the instance argument now?

Reply:
My guess is yes because agent 5 is on a separate branch